pkg/nodeagent/controller/operatingsystemconfig/changes.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
```

Comment:
This functionality is not tested explicitly since it's "internal" to this package and just utility. The integration test should verify the logic entirely. Are we missing a case that you'd like to be covered?

Reply:
I did not see the integration test yet, when I was checking this commit. 
Atm. a test for deleting a dropin is missing, otherwise it is fine.